rllib/policy/tests/test_multi_agent_batch.py

```diff
@@ -0,0 +1,241 @@
+            for i, j in zip(
+                sliced_ma_batches,
+                [
+                    MultiAgentBatch(
+                        {
+                            "0": SampleBatch(
+                                {
+                                    SampleBatch.T: [0],
+                                    SampleBatch.EPS_ID: [0],
+                                    SampleBatch.AGENT_INDEX: [0],
+                                    SampleBatch.SEQ_LENS: [1],
+                                }
+                            )
+                        },
+                        1,
+                    ),
+                    MultiAgentBatch(
+                        {
+                            "0": SampleBatch(
+                                {
+                                    SampleBatch.T: [1],
+                                    SampleBatch.EPS_ID: [0],
+                                    SampleBatch.AGENT_INDEX: [0],
+                                    SampleBatch.SEQ_LENS: [1],
+                                }
+                            )
+                        },
+                        1,
+                    ),
+                    MultiAgentBatch(
+                        {
+                            "1": SampleBatch(
+                                {
+                                    SampleBatch.T: [0],
+                                    SampleBatch.EPS_ID: [1],
+                                    SampleBatch.AGENT_INDEX: [1],
+                                    SampleBatch.SEQ_LENS: [1],
+                                }
+                            )
+                        },
+                        1,
+                    ),
+                    MultiAgentBatch(
+                        {
+                            "1": SampleBatch(
+                                {
+                                    SampleBatch.T: [1],
+                                    SampleBatch.EPS_ID: [1],
+                                    SampleBatch.AGENT_INDEX: [1],
+                                    SampleBatch.SEQ_LENS: [1],
+                                }
+                            )
+                        },
+                        1,
+                    ),
+                ],
+            )
+        ]
+
+    def test_timeslices_partially_overlapping_experiences(self):
+        """Tests if timeslices works as expected on a MultiAgentBatch
+        consisting of two partially overlapping SampleBatches.
+        """
+
+        def _generate_data(agent_idx, t_start):
+            batch = SampleBatch(
+                {
+                    SampleBatch.T: [t_start, t_start + 1],
+                    SampleBatch.EPS_ID: [0, 0],
+                    SampleBatch.AGENT_INDEX: 2 * [agent_idx],
+                    SampleBatch.SEQ_LENS: [2],
+                }
+            )
+            return batch
+
+        policy_batches = {str(idx): _generate_data(idx, idx) for idx in (range(2))}
```

Comment:
nit, just a suggestion, not recommending you to change anything.
since policy_batches only has 2 SampleBatches, it may be clearer to just write them out, instead of calling a _generate_data. it's not that many lines of code anyways.
then, we don't have to imagine what policy_batches look like.
but this is totally fine, so don't change anything. just a random suggestion.

Reply:
Thanks for taking the time to write such suggestions! Code clarity matter :)